lib/que/locker.rb

```diff
@@ -266,6 +266,7 @@ def startup
         CURRENT_HOSTNAME,
         !!@listener,
         "{\"#{@queue_names.join('","')}\"}",
+        Que.major_version,
```

Comment:
I really like this change but would not couple the data version to the Que major version. Sure, if you change the data version you'd want to bump the major version but the reverse might not be true - it's not inconceivable to want to bump the major version without changing the data version. I'm very much in favour of having it as a defined constant that can be easily referenced, I think `lib/que/version.rb` is a good place for it (though you could create a `lib/que/data_version.rb` or similar) but I think conflating the two versions now might cause trouble down the road.

Reply:
Yeah, I have been feeling more and more uncomfortable about this coupling too. I think it's an easy change to make.

Reply:
Ok. What name would you suggest? We should probably rename the `que_version` columns to match.

Reply:
`data_schema_version` or something along those lines.

Reply:
It's tricky calling this a schema version because it suggests that it's something that can/ought to be considered every time there is a schema change. It's really much more limited that than in that it's only useful for additive changes to the `que_jobs` table.

Reply:
`job_schema_version` feels closer?